components/core/src/clp/GlobalSQLiteMetadataDB.cpp

```diff
@@ -175,6 +176,41 @@ get_archives_for_file_select_statement(SQLiteDB& db, string const& file_path) {
 
     return statement;
 }
+
+SQLitePreparedStatement
+get_file_split_statement(SQLiteDB& db, string const& file_orig_id, size_t message_ix) {
+    auto statement_string = fmt::format(
+            "SELECT DISTINCT {}.{}, {}.{} FROM {} JOIN {} ON {}.{} = {}.{} WHERE {}.{} = ?"
+            " AND ? >= {}.{} AND ? < ({}.{} + {}.{}) ORDER BY {} ASC, {} ASC",
+            streaming_archive::cMetadataDB::ArchivesTableName,
+            streaming_archive::cMetadataDB::Archive::Id,
+            streaming_archive::cMetadataDB::FilesTableName,
+            streaming_archive::cMetadataDB::File::Id,
+            streaming_archive::cMetadataDB::ArchivesTableName,
+            streaming_archive::cMetadataDB::FilesTableName,
+            streaming_archive::cMetadataDB::ArchivesTableName,
+            streaming_archive::cMetadataDB::Archive::Id,
+            streaming_archive::cMetadataDB::FilesTableName,
+            streaming_archive::cMetadataDB::File::ArchiveId,
+            streaming_archive::cMetadataDB::FilesTableName,
+            streaming_archive::cMetadataDB::File::OrigFileId,
+            streaming_archive::cMetadataDB::FilesTableName,
+            streaming_archive::cMetadataDB::File::BeginMessageIx,
+            streaming_archive::cMetadataDB::FilesTableName,
+            streaming_archive::cMetadataDB::File::NumMessages,
+            streaming_archive::cMetadataDB::FilesTableName,
+            streaming_archive::cMetadataDB::File::BeginMessageIx,
+            streaming_archive::cMetadataDB::Archive::CreatorId,
+            streaming_archive::cMetadataDB::Archive::CreationIx
+    );
+    SPDLOG_DEBUG("{}", statement_string);
+    auto statement = db.prepare_statement(statement_string.c_str(), statement_string.length());
+    statement.bind_text(1, file_orig_id, true);
+    statement.bind_int(2, static_cast<int>(message_ix));
+    statement.bind_int(3, static_cast<int>(message_ix));
```

Comment:
I think in `WHERE` clauses, you can use arbitrary numeric numbers like what we did in unit test: https://github.com/y-scope/clp/blob/main/components/core/tests/test-SQLiteDB.cpp#L215
By doing this, you don't have to bind `message_idx` twice